src/app/frontend/replicasetdetail/replicasetdetail_state.js

```diff
@@ -52,5 +52,21 @@ export default function stateConfig($stateProvider) {
     controllerAs: 'ctrl',
     url: '/replicasets/:namespace/:replicaSet',
     templateUrl: 'replicasetdetail/replicasetdetail.html',
+    resolve: {
+      replicaSetDetail: resolveDetails,
+    },
   });
 }
+
+
+/**
+ * @param {!StateParams} $stateParams
+ * @param {!angular.$resource} $resource
+ * @ngInject
+ */
+function resolveDetails($stateParams, $resource) {
+  /** @type {!angular.Resource<!backendApi.ReplicaSetDetail>} */
+  let resource = $resource('/api/replicasets/:namespace/:replicaSet', $stateParams);
+
+  return resource.get().$promise;
+}
```

Comment:
@maciaszczykm has already created PR with external resolver file for this.
https://github.com/maciaszczykm/dashboard/commit/7fe92359ec53e61afc2e274259f516234551997b
Maybe this PR can be merged first and then this function can be moved.